scipy/stats/_stats_py.py

```diff
@@ -10675,6 +10675,148 @@ def first_order(t):
     return res.root
 
 
+def lmoments_iv(sample, order, axis, sorted, standardize):
+
+    sample = np.asarray(sample)
+    message = "`sample` must be an array of real numbers."
+    if np.issubdtype(sample.dtype, np.integer):
+        sample = sample.astype(np.float64)
+    if not np.issubdtype(sample.dtype, np.floating):
+        raise ValueError(message)
+
+    message = "`order` must be an array of positive integers."
+    order = np.asarray(order)
+    if (not np.issubdtype(order.dtype, np.integer) or np.any(order <= 0)):
+        raise ValueError(message)
+
+    axis = np.asarray(axis)[()]
+    message = "`axis` must be an integer."
+    if not np.issubdtype(axis.dtype, np.integer) or axis.ndim != 0:
+        raise ValueError(message)
+
+    sorted = np.asarray(sorted)[()]
+    message = "`sorted` must be True or False."
+    if not np.issubdtype(sorted.dtype, np.bool_) or sorted.ndim != 0:
+        raise ValueError(message)
+
+    standardize = np.asarray(standardize)[()]
+    message = "`standardize` must be True or False."
+    if not np.issubdtype(standardize.dtype, np.bool_) or standardize.ndim != 0:
+        raise ValueError(message)
+
+    sample = np.moveaxis(sample, axis, -1)
+    sample = np.sort(sample, axis=-1) if not sorted else sample
+
+    return sample, order, axis, sorted, standardize
+
+
+def _br(x, *, r=0):
+    n = x.shape[-1]
+    x = np.expand_dims(x, axis=-2)
+    x = np.broadcast_to(x, x.shape[:-2] + (len(r), n))
+    x = np.triu(x)
+    j = np.arange(n, dtype=x.dtype)
+    n = np.asarray(n, dtype=x.dtype)[()]
+    return (np.sum(special.binom(j, r[:, np.newaxis])*x, axis=-1)
+            / special.binom(n-1, r) / n)
+
+
+def _prk(r, k):
+    return (-1)**(r-k)*special.binom(r, k)*special.binom(r+k, k)
```

Comment:
I don't have any preferences on the lhs vs rhs of eqn 27 here but don't we want `special.comb` with 
`exact=True` ?

For the small values of r,k I suspect it may not matter much but for larger values it might

Reply:
Yup, that sounds like a possible enhancement to study when `comb` supports arrays.

Reply:
I've taken the liberty of resolving comments to which you gave a thumbs up @rlucas7. I'll also resolve some of the others that are unambiguously addressed.